### PR TITLE
DE2903-filter-out-past-events-for-group-sign-ups

### PR DESCRIPTION
**Pull request by abeeku23**
Updated the code to filter out past events

### CODE REVIEW

File: Gateway/crds-angular.test/Services/GroupServiceTest.cs [hunk 1]
```diff
@@ -387,7 +387,9 @@ public void shouldAddParticipantsToCommunityGroupAndEvents()
                 new MpEvent {EventId = 555},
                 new MpEvent {EventId = 444}
             };
-            groupRepository.Setup(mocked => mocked.getAllEventsForGroup(456, null, false)).Returns(events);
+
+            //groupRepository.Setup(mocked => mocked.getAllEventsForGroup(456, null,false)).Returns(events); DE2903: changed to test current datetime
+            groupRepository.Setup(mocked => mocked.getAllEventsForGroup(456, DateTime.Today, false)).Returns(events);
```
Comment (chris-branch) [inline Gateway/crds-angular.test/Services/GroupServiceTest.cs]: Can we replace DateTime.Today with a mocked DataTime wrapper (see IDateTime) so our tests aren't dependent on the current system clock?

File: Gateway/crds-angular/Services/GroupService.cs [hunk 2]
```diff
@@ -57,7 +59,21 @@ public class GroupService : IGroupService
         private readonly int _groupAgeRangeAttributeTypeId;
         private readonly int _groupRoleLeader;
         private readonly int _domainId;
-
+        private IGroupRepository object1;
```
Comment (chris-branch) [inline Gateway/crds-angular/Services/GroupService.cs]: Was 62-76 added by accident?  If so, lets delete.

File: Gateway/crds-angular/Services/GroupService.cs [hunk 3]
```diff
@@ -103,8 +120,28 @@ public class GroupService : IGroupService
             _smallGroupTypeId = _configurationWrapper.GetConfigIntValue("SmallGroupTypeId");
             _onsiteGroupTypeId = _configurationWrapper.GetConfigIntValue("OnsiteGroupTypeId");
             _childcareEventTypeId = _configurationWrapper.GetConfigIntValue("ChildcareEventType");
+            _dateTimeWrapper = dateTimeWrapper;
+
 
+        }
 
+        public GroupService(IGroupRepository object1, IConfigurationWrapper object2, IEventRepository object3, IContactRelationshipRepository object4, IServeService object5, IParticipantRepository object6, ICommunicationRepository object7, IContactRepository object8, IObjectAttributeService object9, IApiUserRepository object10, IAttributeRepository object11, IEmailCommunication object12, IUserRepository object13, IInvitationRepository object14, IAttributeService object15)
```
Comment (chris-branch) [inline Gateway/crds-angular/Services/GroupService.cs]: Was 128-144 added by accident?  If so, lets delete.

Comment (abeeku23) [inline Gateway/crds-angular/Services/GroupService.cs]: I think I clicked through some preview changes to fix some errors. Removed and will rebuild ASAP

**Pull request merged**